src/fsharp/CompileOps.fs

```diff
@@ -2549,6 +2547,56 @@ let GetAutoOpenAttributes ilg ilModule =
 
 let GetInternalsVisibleToAttributes ilg ilModule = 
     ilModule |> GetCustomAttributesOfILModule |> List.choose (TryFindInternalsVisibleToAttr ilg)
+
+let findPrimaryAssembly (data: TcConfigBuilder) =
+    let assemblies =
+        data.referencedDLLs
+        |> List.map (fun r -> r.Text)
+
+    let readerSettings: ILReaderOptions = 
+        { pdbDirPath=None
+          ilGlobals = EcmaMscorlibILGlobals
+          reduceMemoryUsage = data.reduceMemoryUsage
+          metadataOnly = MetadataOnlyFlag.Yes
+          tryGetMetadataSnapshot = data.tryGetMetadataSnapshot }
+
+    let nameSystem = ["System"]
+    let nameObject = "Object"
+
+    let hasNoPiaLocalTypes (reader: ILModuleReader) =
+        reader.ILModuleDef.TypeDefs.AsArray
+        |> Array.exists (fun (ilTypeDef: ILTypeDef) ->
+            ilTypeDef.CustomAttrs.AsArray
+            |> Array.exists (fun attr -> attr.Method.MethodRef.DeclaringTypeRef.Name = "System.Runtime.InteropServices.TypeIdentifierAttribute"))
```

Comment:
O(n^2) - are these arrays small?

Reply:
Also: do we need to do AsArray? What's the underlying type? Seq.exists may be an alternative here.

Reply:
The underlying type is an array and these arrays will be small. It's fine that we iterate like this.

IMO, this should be an `ImmutableArray`.